hack/run-test-must-gather-e2e.sh

```diff
@@ -1,15 +1,176 @@
-#! /usr/bin/env bash
+#!/usr/bin/env bash
 
 source hack/common.sh
 
+### must-gather suite runner
+# we expect the lane to run against a already configured cluster
+# we use this verbose form to play nice with envsubst
+if [ -z "${SETUP}" ]; then
+	SETUP="false"
+fi
+if [ -z "${TEARDOWN}" ]; then
+	TEARDOWN="false"
+fi
+if [ -z "${RUN_TESTS}" ]; then
+	RUN_TESTS="true"
+fi
+
 NO_COLOR=""
 if ! which tput &> /dev/null 2>&1 || [[ $(tput -T$TERM colors) -lt 8 ]]; then
-  echo "Terminal does not seem to support colored output, disabling it"
+  echo "Terminal does not seem to support colored output, disabling it" 1>&2
   NO_COLOR="-ginkgo.no-color"
 fi
 
-setupreport
+if [ -n "${E2E_MUST_GATHER_FOCUS}" ]; then
+	FOCUS="-ginkgo.focus=${E2E_MUST_GATHER_FOCUS}"
+fi
+if [ -n "${E2E_MUST_GATHER_SKIP}" ]; then
+	SKIP="-ginkgo.skip=${E2E_MUST_GATHER_SKIP}"
+fi
+
+DRY_RUN="false"
+REPORT_FILE=""
+
+# so few arguments is no bother enough for getopt
+while [[ $# -gt 0 ]]; do
+	case "$1" in
+		--setup)
+			SETUP="true"
+			shift
+			;;
+		--teardown)
+			TEARDOWN="true"
+			shift
+			;;
+		--no-run-tests)
+			RUN_TESTS="false"
+			shift
+			;;
+		--no-color)
+			NO_COLOR="--ginkgo.no-color"
+			shift
+			;;
+		--dry-run)
+			DRY_RUN="true"
+			shift
+			;;
+		--focus)
+			FOCUS="-ginkgo.focus='$2'"
+			shift
+			shift
+			;;
+		--skip)
+			SKIP="-ginkgo.skip='$2'"
+			shift
+			shift
+			;;
+		--report-file)
+			REPORT_FILE="$2"
+			shift
+			shift
+			;;
+		--help)
+			echo "usage: $0 [options]"
+			echo "[options] are always '--opt val' and never '--opt=val'"
+			echo "available options:"
+			echo "--setup               perform cluster setup before to run the tests (if enabled, see --no-run-tests)"
+			echo "--teardown            perform cluster teardown after having run the tests (if enabled, see --no-run-tests)"
+			echo "--no-run-tests        don't run the tests (see --setup and --teardown)"
+			echo "--no-color            force colored output to off"
+			echo "--dry-run             logs what about to do, but don't actually do it"
+			echo "--focus <regex>       only run cases matching <regex> (passed to -ginkgo.focus)"
+			echo "--skip <regex>        skip cases matching <regex> (passed to -ginkgo.skip)"
+			echo "--report-file <file>  write report file for this suite on <file>"
+			echo "--help                shows this message and helps correctly"
+			exit 0
+			;;
+		*)
+			echo "unrecognized option: $1"
+			echo "use the form --opt val and not the form --opt=val"
+			exit 1
+			;;
+	esac
+done
+
+if [ -z "${REPORT_FILE}" ]; then
+	REPORT_FILE="${REPORT_DIR}/e2e-must-gather-run.xml"
+fi
+
+function setup() {
+	if [[ "${SETUP}" != "true" ]]; then
+		return 0
+	fi
```

Comment:
I understand this is how we've been doing it for serial suite, but why do we need it? don't we limit our selves here? what if we see the need for other new ginkgo flags? 
If we want to execute other commands that aren't supported by this, we'll be banned
https://github.com/openshift-kni/numaresources-operator/pull/686/files#diff-63231dd0167658c2ecd90e27f927a8312a61ac0849cec95b8fe72b1263a06360R88
do we really need it that strict? 
plus I see the tests binaries are copied under the image:
https://github.com/openshift-kni/numaresources-operator/pull/686/files#diff-1d0fed26e7daaae4aa24a5ed9bab7c25d82cac3a51199b90cf9c73e01da4b5b8R10 so why not enable using them directly on new container?

Reply:
just to clarify my point, what I think - and might be mistaken- is that if we had another entrypoint (not command) like bin/sh and we do not limit the command to be within the ginkgo flags specified above, then I think we wouldn't be doing or needing this change now since the tests binaries are in fact shipped into the image (an till now we use only the serial suite .test).